### ISSUE TITLE
Bug: Screen reader announces "Not selected" for non-selectable list items (selection-mode="none")

### ISSUE THREAD

**Issue by yuhyun-kwon** [Esri/calcite-components#14519]
**monday.com sync:** #12131099086

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Which Component

List Item

### Actual Behavior

Screen readers announce list items with `selection-mode="none"` as "Not selected", implying they are selectable.

### Expected Behavior

- List items should not be announced as selectable when selection-mode="none" is set.
- Screen readers should not announce selection state (e.g., "selected" / "not selected") for non-selectable items.

### Reproduction Sample

https://codepen.io/DintaMel-the-lessful/pen/MYbORaV?editors=100

### Reproduction Steps

1. Open the Codepen sample with List component
2. Turn on NVDA screen reader
3. Hear what it announces

### Reproduction Version

5.0

### Relevant Info

It seems to happen in NVDA but not on Mac Voice over.

### Priority impact

impact - p3 - not time sensitive

### Impact

Screen reader users could be confused when they hear 'not selected' for the list item that is not selectable.

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/calcite-ui-icons
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Online
